### PR TITLE
Push images to ECR, then we'll setup Harbor as proxy cache

### PR DESCRIPTION
**Pull request by bebosudo**
As requested by Paolo, let's use ECR, to reduce SPOF problems possibly generated by Harbor.

The action triggered by opening this PR will build the image, but won't push it, so it'll be enough to test whether the new authentication system works. A temporary test commit was added, but will be force-removed before marking this PR as ready

### CODE REVIEW

**Comment by bebosudo** [comment]
All clear and ready for review @pabloaledo!

File: .github/workflows/build-containers.yml [hunk 1]
```diff
@@ -101,5 +113,5 @@ jobs:
           cache-from: type=gha
           cache-to: type=gha,mode=max
           tags: |
-            cr.seqera.io/spack/${{ env.container }}
-            cr.seqera.io/spack/${{ env.versioned }}
+            195996028523.dkr.ecr.eu-west-1.amazonaws.com/wave/spack/${{ env.container }}
+            195996028523.dkr.ecr.eu-west-1.amazonaws.com/wave/spack/${{ env.versioned }}
```
Comment (pabloaledo) [inline .github/workflows/build-containers.yml:117]: Maybe instead of setting Harbor as a proxy, can we push here both to Harbor and ECR?. Easy :smile:

**Review by pabloaledo — APPROVED** [review]

**Comment by pabloaledo** [comment]
image

As developer, I don't have access to the ECR registry, I think it would be nice to use another registry/change the policies to be able to see and use the generated image

File: .github/workflows/build-containers.yml (hunk 1 above)
Comment (bebosudo) [reply .github/workflows/build-containers.yml:117]: Once the proxy repo in Harbor is set up, you'll be able to pull from Harbor without needing any permissions to access ECR.

**Comment by bebosudo** [comment]
I manually ran the pipeline to push to the new location, but I forgot to create the `/ubuntu-jammy` sub-URL of the container image repo `wave/spack/ubuntu-jammy` in ECR. Manually created and now we'll push images.
I'll investigate a way to create the repo automatically in the CI.

**Pull request merged**